File: ui/src/app/edge/settings/systemupdate/getSystemUpdateStateRequest.ts
import { JsonrpcRequest } from "src/app/shared/jsonrpc/base";

/**
 * Represents a JSON-RPC Request to get the current state of system update on OpenEMS Edge.
 * 
 * <pre>
 * {
 *   "jsonrpc": "2.0",
 *   "id": "UUID",
 *   "method": "getSystemUpdateState",
 *   "params": {
 *   }
 * }
 * </pre>
 */
export class GetSystemUpdateStateRequest extends JsonrpcRequest {

    static METHOD: string = "getSystemUpdateState";

    public constructor(
    ) {
        super(GetSystemUpdateStateRequest.METHOD, {});
    }

}

File: io.openems.edge.core/src/io/openems/edge/core/host/jsonrpc/GetSystemUpdateStateRequest.java

package io.openems.edge.core.host.jsonrpc;

import com.google.gson.JsonObject;

import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
import io.openems.common.jsonrpc.base.JsonrpcRequest;
import io.openems.common.utils.JsonUtils;

/**
 * Gets the System Update State.
 *
 * <pre>
 * {
 *   "jsonrpc": "2.0",
 *   "id": "UUID",
 *   "method": "getSystemUpdateState",
 *   "params": {
 *   	"debug"?: boolean
 *   }
 * }
 * </pre>
 */
public class GetSystemUpdateStateRequest extends JsonrpcRequest {

	public static final String METHOD = "getSystemUpdateState";

	/**
	 * Parses a generic {@link JsonrpcRequest} to a
	 * {@link GetSystemUpdateStateRequest}.
	 *
	 * @param r the {@link JsonrpcRequest}
	 * @return the {@link GetSystemUpdateStateRequest}
	 * @throws OpenemsNamedException on error
	 */
	public static GetSystemUpdateStateRequest from(JsonrpcRequest r) throws OpenemsNamedException {
		return new GetSystemUpdateStateRequest(r);
	}

	public GetSystemUpdateStateRequest() {
		super(METHOD);
	}

	private GetSystemUpdateStateRequest(JsonrpcRequest request) {
		super(request, METHOD);
	}

	@Override
	public JsonObject getParams() {
		return JsonUtils.buildJsonObject() //
				.build();
	}

}
